external/llvm-project/mlir/lib/Dialect/Tosa/Transforms/TosaPartition.cpp

```diff
@@ -360,53 +391,95 @@ void outlineConvPartOps(Operation *convOp, ArrayRef<Operation *> secondOps,
   }
 
   // Erase the ops we outlined, which should be safe now.
-  for (auto &op : llvm::make_early_inc_range(llvm::reverse(secondOps))) {
+  for (auto &op : llvm::make_early_inc_range(llvm::reverse(trailingOps))) {
     if (op->use_empty()) {
       op->erase();
     }
   }
-  assert(convOp->use_empty() && "expected 'op' to have no uses");
-  convOp->erase();
-  for (auto &op : llvm::make_early_inc_range(frontOps)) {
+  assert(anchorOp->use_empty() && "expected 'op' to have no uses");
+  anchorOp->erase();
+  for (auto &op : llvm::make_early_inc_range(leadingOps)) {
     if (op->use_empty()) {
       op->erase();
     }
   }
 }
 
+} // namespace
+
+namespace mlir {
+namespace tosa {
+
+class PartitionConfig {
+public:
+  virtual bool isAnchorOp(Operation *) = 0;
+  virtual bool isLeadingOp(Operation *) = 0;
+  virtual bool isTrailingOp(Operation *) = 0;
+  virtual std::string attributeName() = 0;
+  virtual ~PartitionConfig() = default;
+};
+
+class SimpleDefaultPartitionConfig : public PartitionConfig {
+public:
+  bool isAnchorOp(Operation *op) override { return isa<tosa::Conv2DOp>(op); }
+  bool isLeadingOp(Operation *op) override {
+    return isConstantZero(op) || isFusibleOp(op);
+  }
+  bool isTrailingOp(Operation *op) override { return isFusibleOp(op); }
+  std::string attributeName() override { return "kernel"; }
+};
+
+class PartitionConfigFromOptions : public PartitionConfig {
```

Comment:
A configuration based on the command-line sub-options (defined in Passes.td).  The creator of the configuration object will look at the Option objects and pass their values in to the constructor, which saves them in the object.  Any defaulting happens in the caller.